cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -366,42 +411,60 @@ func (m *azureCache) FindForInstance(instance *azureRef, vmType string) (cloudpr
 		return nil, err
 	}
 	inst := azureRef{Name: resourceID}
+
+	if unowned, err := m.isUnownedInstance(inst, ""); unowned || err != nil {
+		return nil, err
+	}
+
+	if invalid, err := m.isInvalidVMType(inst, vmType, vmsPoolSet); invalid || err != nil {
+		return nil, err
+	}
+
+	// Look up caches for the instance.
+	klog.V(6).Infof("FindForInstance: attempting to retrieve instance %v from cache", m.instanceToNodeGroup)
+	if nodeGroup := m.getInstanceFromCache(inst.Name); nodeGroup != nil {
+		klog.V(4).Infof("FindForInstance: found node group %q in cache", nodeGroup.Id())
+		return nodeGroup, nil
+	}
+	klog.V(4).Infof("FindForInstance: Couldn't find node group of instance %q", inst)
+	return nil, nil
+}
+
+func (m *azureCache) isUnownedInstance(inst azureRef, reason string) (bool, error) {
```

Comment:
seems like calling this function is sufficient and HasInstance/FindForInstance do not need to also do their own `m.unownedInstances[inst]` check

Reply:
thats true more of a sanity check, we can probably remove that. 

unownedInstances isn't a very good way to track this anyway as it doesn't really represent all unowned instances.

Reply:
> more of a sanity check

Or at least maybe there should be a comment about it if it is not too intuitive